File: src/main/java/io/pageobject/generator/locator/protractor/NgBindHtmlLocatorSource.java
package io.pageobject.generator.locator.protractor;

import io.pageobject.generator.GeneratorContext;
import io.pageobject.generator.attribute.HtmlAttributes;
import io.pageobject.generator.locator.AbstractLocatorSource;
import org.jsoup.nodes.Element;

import static io.pageobject.generator.attribute.Attributes.extractAttributeValue;

public class NgBindHtmlLocatorSource extends AbstractLocatorSource {

    @Override
    public String extractLocatorValue(Element element, GeneratorContext context) {
        return extractAttributeValue(element, HtmlAttributes.NG_BIND_HTML);
    }

    public String locatorTemplate() {
        return "by.exactBinding('%s')";
    }

}

File: lib/locator/protractor/ng-bind-html-locator-source.js
"use strict";

const AbstractLocatorSource = require('../abstract-locator-source');
const _ = require('lodash');
const extractAttributeValue = require('../../attribute/attributes').extractAttributeValue;
const HtmlAttribute = require('../../attribute/html-attribute');

const template = _.template("by.exactBinding('<%= value %>')");

class NgBindHtmlLocatorSource extends AbstractLocatorSource {

  extractLocatorValue(context) {
    return extractAttributeValue(context.element, HtmlAttribute.NG_BIND_HTML);
  }

  locatorTemplate() {
    return template;
  }

}

module.exports = NgBindHtmlLocatorSource;